ui-tests/src/test/java/org/openmrs/reference/AddRolesToUserTest.java

```diff
@@ -42,9 +44,6 @@ public void addRolesToUserTest() throws InterruptedException {
             manageUserPage.assignRolesToUser(oldRole, role.getKey(), NURSE_USERNAME);
 
             reLoginAsUser();
-            if (homePage.numberOfAppsPresent() != role.getValue()) {
-                throw new AssertionError("role " + role + " doesn't have matching number of accessible applications: should be:" + role.getValue() + "is:" + homePage.numberOfAppsPresent());
```

Comment:
we need the check atleast to check the number of roles given as per login and number of apps present, if you remove the if statement the whole meaning will have changed the  function.